include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -243,6 +244,32 @@ inline ConstMemoryInfo AllocatorImpl<T>::GetInfo() const {
   return ConstMemoryInfo{out};
 }
 
+template <typename T>
+inline std::unordered_map<std::string, std::string> AllocatorImpl<T>::GetStats() const {
+  AllocatorWithDefaultOptions allocator;
+  char* raw_stats = nullptr;
+  ThrowOnError(GetApi().AllocatorGetStats(this->p_, allocator, &raw_stats));
```

Comment:
>ThrowOnError(GetApi().AllocatorGetStats(this->p_, allocator, &raw_stats)); [](http://example.com/codeflow?start=2&length=74)

Most allocators do not implement GetStats() and it would be tedious for the user handle the exception.

Suggest to rework this interface to return Ort::Status.
There is an utility function that converts OrtStatus to Ort::Status.

Clearly document that error code would be set to ORT_NOT_IMPLEMENTED.